### ISSUE TITLE
Fix `at-rule-no-unknown` false positives for stage 2 conditional at-rules

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#8396]
### What minimal example or steps are needed to reproduce the bug?

```css
@when media(width >= 640px) and (supports(display: flex) or supports(display: grid)) {
  /* A */
} @else media(pointer: coarse) {
  /* B */
} @else {
  /* C */
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "at-rule-no-unknown": true
  }
}
```

### How did you run Stylelint?

[demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACB3AFjAdgAgLYygEsBDAChUKgBc1sA+AXmwDYAWABgAcAPASm2KYo2UgGcArp04QATlVGkiozgBtiAT3jYAZiph9ss7BKmz5iwsrWbsAcxmVe-YAB0c2APQAqbAEFsXh5uAL7YSDAqojB4BCSk0oSYVDAyWpDEMlHObtiePgBCAUGYoeGR0a7u3tgAwkUhIAA0INqEegByxPgIIPpdqjAAdGCiok3gEJittj2VuS4gMuJ6ogtac7nYC8RUALRLeruYELvimADWxyiYa9hUSzA52MENzZBThLYAYrK4Oz0AK1Ek3GsE4Y0QGwWoio6j0KkSVFuCzUyRhC0aT2hsPhiN272muxhgigGSgyJAqJg6JADWCQA)

### Details
As a precedent we already have a stage 1 at-rule: [`@custom-selector`](https://cssdb.org/#custom-selectors).

https://www.w3.org/TR/css-conditional-5/
https://cssdb.org/#when-else-rules
https://caniuse.com/css-when-else
https://github.com/w3c/csswg-drafts/commit/0a8f9f90f0095c17328dfccd9eacda77d66ba693

### What did you expect to happen?

no errors

### What actually happened?

Unexpected unknown at-rule "@when"
Unexpected unknown at-rule "@else" × 2

### Do you have a proposal to fix the bug?

1. add `'else'` and `'when'` to https://github.com/stylelint/stylelint/blob/9f393c5dc5d88ddd7eb2690b6c875810defbf162/lib/reference/atKeywords.mjs#L47
2. add `'if'` to https://github.com/stylelint/stylelint/blob/9f393c5dc5d88ddd7eb2690b6c875810defbf162/lib/reference/atKeywords.mjs#L3

**Comment by romainmenke** [comment]
I don't think we should add these at this time.

There are no implementations, no polyfills and no CSS author facing documentation.
The CSSWG is also unsure what the name should be and can't decide between `@when` and `@if`, effectively blocking any progress.

Is this something anyone is actually writing?

**Comment by Mouvedia** [comment]
> There are no implementations, no polyfills and no CSS author facing documentation.

That's why I created an issue instead of a PR.

> Maybe we should document this better?

This is covered by the planned tasks resulting from the discussion in #7510.
I don't have the time lately, but it will eventually be documented.

> as they can't decide between `@when` and `@if`

Are you talking about w3c/csswg-drafts#6684?

**Comment by romainmenke** [comment]
> > as they can't decide between `@when` and `@if`
> Are you talking about https://github.com/w3c/csswg-drafts/issues/6684?

Yes

**Comment by Mouvedia** [comment]
see https://github.com/stylelint/stylelint/issues/7510#issuecomment-2661583197

There is too much uncertainty on which of the two at-rules will eventually become the deprecated one.
